contrib/pg_tde/documentation/docs/setup.md

```diff
@@ -173,16 +155,36 @@ Here's how to do it:
         This setup is intended for development and stores the keys unencrypted in the specified data file.    
 
         ```sql
-        SELECT pg_tde_add_key_provider_file('provider-name','/path/to/the/keyring/data.file');
+        SELECT pg_tde_add_global_key_provider_file('provider-name','/path/to/the/keyring/data.file');
         ```
 
-4. Rotate the principal key. Don't forget to specify the `PG_TDE_GLOBAL` constant to rotate only the principal key for WAL.
+3. Create principal key
+    
+    ```sql
+    SELECT pg_tde_set_server_principal_key('principal-key', 'provider-name');
+    ```
+
+4. Enable WAL level encryption using the `ALTER SYSTEM` command. You need the privileges of the superuser to run this command:
 
     ```sql
-    SELECT pg_tde_rotate_principal_key('PG_TDE_GLOBAL', 'new-principal-key', 'provider-name');
+    ALTER SYSTEM SET pg_tde.wal_encrypt = on;
     ```
 
-Now all WAL files are encrypted for both encrypted and unencrypted tables.
+5. Restart the server to apply the changes.
+
+    * On Debian and Ubuntu:    
+
+       ```sh
+       sudo systemctl restart postgresql.service
+       ```
+    
+    * On RHEL and derivatives
+
+       ```sh
+       sudo systemctl restart postgresql-17
+       ```
+
+Now WAL files start to be encrypted for both encrypted and unencrypted tables.
```

Comment:
for everything, not just tables.

Reply:
I meant "encrypted and unencrypted tables with user data". Meaning if a user has dbs with pg_tde created against them and those without pg_tde, WAL will be encrypted for all of them

Reply:
WAL is not specific to tables, any database operation possibly can be logged there. For example, when we create an internal key, we write it into the WAL. Now it will be encrypted, and it's not a table. Everything that is written into the WAL will be encrypted.

Reply:
```
Now all data in WAL files start to be encrypted, including data for unencrypted table.
```
?